client-lite/src/exe/docs.cpp

```diff
@@ -16,57 +16,95 @@ namespace msdoutil = microsoft::deliveryoptimization::util::details;
 
 #include "config_manager.h"
 #include "download_manager.h"
+#include "do_noncopyable.h"
 #include "proc_launch_helper.h"
 #include "rest_http_controller.h"
 #include "rest_port_advertiser.h"
 #include "trace_sink.h"
 
 using namespace std::chrono_literals; // NOLINT(build/namespaces) how else should we use chrono literals?
 
-class ProcessController
+class ProcessController : DONonCopyable
```

Comment:
>: DONonCopyable [](start = 24, length = 15)

Perhaps there is no need to make this a singleton as there are no non-static members

Reply:
Non-copyable doesn't make it a singleton because you can create multiple instances of this class.

---
In reply to: [595354139](https://github.com/microsoft/do-client/pull/11#discussion_r595354139) [](ancestors = 595354139)

Reply:
Ah right, there's a convention for singleton involving static instance of the class itself and a private constructor

---
In reply to: [595411879](https://github.com/microsoft/do-client/pull/11#discussion_r595411879) [](ancestors = 595411879,595354139)